tools/run_tests/jobset.py

```diff
@@ -211,6 +223,8 @@ def state(self, update_cache):
       # implemented efficiently. This is an experiment to workaround the issue by making sure
       # results.xml file is small enough.
       filtered_stdout = filtered_stdout[-128:]
+      self.result.message = filtered_stdout
+      self.result.elapsed_time = elapsed
       if self._xml_test is not None:
```

Comment:
I agree. JobResult allows us to move all reporting stuff to a different util. I can work on this in a different change. I'll work resolve the conflict and work on mako template first.